src/store/LastDigitsStore.js

```diff
@@ -0,0 +1,86 @@
+import { observable, action, computed, reaction } from 'mobx';
+
+export default class LastDigitsStore {
+    constructor(mainStore) {
+        this.mainStore = mainStore;
+        reaction(() => this.mainStore.chart.currentActiveSymbol, this.showLastDigitStats);
```

Comment:
@MahboobehMohammadi it'd be better to use `componentDidUpdate` lifecycle method in `LastDigits` component and check the `marketDisplayName` property to reload the data.
```
componentDidUpdate({ marketDisplayName }) {
    if (this.props.marketDisplayName !== marketDisplayName) {
        this.props.showLastDigitStats();
    }
}
```

Reply:
@morteza-binary what is the difference?

Reply:
@MahboobehMohammadi It's more readable for other developers and makes the maintenance much easier. 
Also in this case, if you need to use reaction you should clear reaction when the component is not mounted and create it after mounting.